src/components/Alerts.js

```diff
@@ -0,0 +1,154 @@
+import { Clickable, Link } from 'src/components/common'
+import { icon } from 'src/components/icons'
+import PopupTrigger from 'src/components/PopupTrigger'
+import colors from 'src/libs/colors'
+import { usePrevious } from 'src/libs/react-utils'
+import { useServiceAlerts } from 'src/libs/service-alerts'
+import * as Utils from 'src/libs/utils'
+
+
+const Alert = ({ alert }) => {
+  const { title, message, link, linkTitle, severity } = alert
+
+  const [baseColor, ariaLabel] = Utils.switchCase(severity,
+    ['success', () => [colors.success, 'success notification']],
+    ['info', () => [colors.accent, 'info notification']],
+    ['welcome', () => [colors.accent, 'welcome notification']],
+    ['warn', () => [colors.warning, 'warning notification']],
+    ['error', () => [colors.danger, 'error notification']],
+    [Utils.DEFAULT, () => [colors.accent, 'notification']]
+  )
+  const iconType = Utils.switchCase(severity,
+    ['success', () => 'success-standard'],
+    ['warn', () => 'warning-standard'],
+    ['error', () => 'error-standard']
+  )
+
+  return div({
+    'data-testid': 'alert',
+    style: {
+      display: 'flex',
+      flexDirection: 'column',
+      padding: '0.75rem 1rem',
+      borderRadius: '4px',
+      backgroundColor: baseColor(0.15),
+      boxShadow: '0 0 4px 0 rgba(0,0,0,0.5)',
+      cursor: 'auto',
+      fontSize: 12
+    }
+  }, [
+    div({ style: { display: 'flex', alignItems: 'center' } }, [
+      !!iconType && icon(iconType, {
+        'aria-hidden': false,
+        'aria-label': ariaLabel,
+        size: 26,
+        style: { color: baseColor(), flexShrink: 0, marginRight: '0.5rem' }
+      }),
+      div({ style: { fontSize: 14, fontWeight: 600, overflowWrap: 'break-word' } }, [title])
+    ]),
+    !!message && div({ style: { marginTop: '0.5rem', fontSize: 12, fontWeight: 500, overflowWrap: 'break-word' } }, [message]),
+    link && div({ style: { marginTop: '0.25rem' } }, [
+      h(Link, { ...Utils.newTabLinkProps, href: link, style: { fontWeight: 700 } }, [
+        linkTitle || 'Read more'
+      ])
+    ])
+  ])
+}
+
+const AlertsList = ({ alerts }) => {
+  return ul({
+    style: {
+      padding: 0,
+      margin: 0,
+      listStyleType: 'none'
+    }
+  }, [
+    _.map(([index, alert]) => li({
+      key: alert.id,
+      style: { marginTop: index === 0 ? 0 : '0.5rem' }
+    }, [
+      h(Alert, { alert })
+    ]), Utils.toIndexPairs(alerts))
+  ])
+}
+
+
+const AlertsIndicator = ({ style }) => {
+  const [open, setOpen] = useState(false)
+  const [animating, setAnimating] = useState(false)
+
+  const alerts = useServiceAlerts()
+
+  const previousAlertIds = usePrevious(_.map('id', alerts))
+  const hasNewAlerts = _.size(_.difference(_.map('id', alerts), previousAlertIds)) > 0
+
+  const numAlerts = _.size(alerts)
+
+  useEffect(() => {
+    if (hasNewAlerts) {
+      setAnimating(true)
+      setTimeout(() => {
+        setAnimating(false)
+      }, 1000)
```

Comment:
You may want to listen for the `animationend` event [1] instead to make this robust in the case of a duration change.
[1] https://developer.mozilla.org/en-US/docs/Web/API/Element/animationend_event

Reply:
Today I learned. Changed in https://github.com/DataBiosphere/terra-ui/pull/3361/commits/f280ac13228f4f614944453e1aeb5cf5dc70dfff.

Reply:
The aforementioned documentation says,

> If the animation aborts before reaching completion, such as if the element is removed from the DOM or the animation is removed from the element, the `animationend` event is not fired.

Do we need to handle such a case?

Reply:
It should be fine. The Alerts component being unmounted is the only reason why that element should be removed from the DOM.